{
  "cve": "CVE-2020-1737",
  "mitre": {
    "cpes": [],
    "created": "2020-03-09T15:11:38+00:00",
    "description": "A flaw was found in Ansible 2.7.17 and prior, 2.8.9 and prior, and 2.9.6 and prior when using the Extract-Zip function from the win_unzip module as the extracted file(s) are not checked if they belong to the destination folder. An attacker could take advantage of this flaw by crafting an archive anywhere in the file system, using a path traversal. This issue is fixed in 2.10.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {},
      "cvssV3_1": {
        "score": 7.5,
        "vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:C/C:H/I:H/A:H"
      },
      "cvssV4_0": {}
    },
    "mitre_repo_path": "cves/2020/1xxx/CVE-2020-1737.json",
    "references": [
      "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1737",
      "https://github.com/ansible/ansible/issues/67795",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FWDK3QUVBULS3Q3PQTGEKUQYPSNOU5M3/",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QT27K5ZRGDPCH7GT3DRI3LO4IVDVQUB7/",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/U3IMV3XEIUXL6S4KPLYYM4TVJQ2VNEP2/",
      "https://security.gentoo.org/glsa/202006-11"
    ],
    "title": null,
    "updated": "2024-08-04T06:46:30.898000+00:00",
    "vendors": [],
    "weaknesses": [
      "CWE-22"
    ]
  },
  "nvd": {
    "cpes": [
      "cpe:2.3:a:redhat:ansible_engine:*:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:ansible_tower:*:*:*:*:*:*:*:*"
    ],
    "created": "2020-03-09T16:15:12.407000+00:00",
    "description": "A flaw was found in Ansible 2.7.17 and prior, 2.8.9 and prior, and 2.9.6 and prior when using the Extract-Zip function from the win_unzip module as the extracted file(s) are not checked if they belong to the destination folder. An attacker could take advantage of this flaw by crafting an archive anywhere in the file system, using a path traversal. This issue is fixed in 2.10.",
    "metrics": {
      "cvssV2_0": {
        "score": 4.6,
        "vector": "AV:L/AC:L/Au:N/C:P/I:P/A:P"
      },
      "cvssV3_0": {},
      "cvssV3_1": {
        "score": 7.8,
        "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
      },
      "cvssV4_0": {}
    },
    "nvd_repo_path": "2020/CVE-2020-1737.json",
    "references": [
      "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1737",
      "https://github.com/ansible/ansible/issues/67795",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FWDK3QUVBULS3Q3PQTGEKUQYPSNOU5M3/",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QT27K5ZRGDPCH7GT3DRI3LO4IVDVQUB7/",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/U3IMV3XEIUXL6S4KPLYYM4TVJQ2VNEP2/",
      "https://security.gentoo.org/glsa/202006-11"
    ],
    "title": null,
    "updated": "2023-11-07T03:19:31.313000+00:00",
    "vendors": [
      "redhat",
      "redhat$PRODUCT$ansible_engine",
      "redhat$PRODUCT$ansible_tower"
    ],
    "weaknesses": [
      "CWE-22"
    ]
  },
  "opencve": {
    "changes": [],
    "cpes": {
      "data": [
        "cpe:/a:redhat:ansible_engine:2.7::el7",
        "cpe:/a:redhat:ansible_engine:2.8::el7",
        "cpe:/a:redhat:ansible_engine:2.8::el8",
        "cpe:/a:redhat:ansible_engine:2.9::el7",
        "cpe:/a:redhat:ansible_engine:2.9::el8",
        "cpe:/a:redhat:ansible_engine:2::el7",
        "cpe:/a:redhat:ansible_engine:2::el8",
        "cpe:/a:redhat:ansible_tower:3.4::el7",
        "cpe:/a:redhat:ansible_tower:3.5::el7",
        "cpe:2.3:a:redhat:ansible_engine:*:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:ansible_tower:*:*:*:*:*:*:*:*"
      ],
      "providers": [
        "nvd",
        "redhat"
      ]
    },
    "created": {
      "data": "2020-02-18T00:00:00+00:00",
      "provider": "redhat"
    },
    "description": {
      "data": "A flaw was found in Ansible 2.7.17 and prior, 2.8.9 and prior, and 2.9.6 and prior when using the Extract-Zip function from the win_unzip module as the extracted file(s) are not checked if they belong to the destination folder. An attacker could take advantage of this flaw by crafting an archive anywhere in the file system, using a path traversal. This issue is fixed in 2.10.",
      "provider": "mitre"
    },
    "metrics": {
      "cvssV2_0": {
        "data": {
          "score": 4.6,
          "vector": "AV:L/AC:L/Au:N/C:P/I:P/A:P"
        },
        "provider": "nvd"
      },
      "cvssV3_0": {
        "data": {},
        "provider": null
      },
      "cvssV3_1": {
        "data": {
          "score": 7.5,
          "vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:C/C:H/I:H/A:H"
        },
        "provider": "mitre"
      },
      "cvssV4_0": {
        "data": {},
        "provider": null
      },
      "kev": {
        "data": {},
        "provider": null
      },
      "ssvc": {
        "data": {},
        "provider": null
      },
      "threat_severity": {
        "data": "Moderate",
        "provider": "redhat"
      }
    },
    "references": {
      "data": [
        "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1737",
        "https://github.com/ansible/ansible/issues/67795",
        "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FWDK3QUVBULS3Q3PQTGEKUQYPSNOU5M3/",
        "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QT27K5ZRGDPCH7GT3DRI3LO4IVDVQUB7/",
        "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/U3IMV3XEIUXL6S4KPLYYM4TVJQ2VNEP2/",
        "https://nvd.nist.gov/vuln/detail/CVE-2020-1737",
        "https://security.gentoo.org/glsa/202006-11",
        "https://www.cve.org/CVERecord?id=CVE-2020-1737"
      ],
      "providers": [
        "mitre",
        "nvd",
        "redhat"
      ]
    },
    "title": {
      "data": "ansible: Extract-Zip function in win_unzip module does not check extracted path",
      "provider": "nvd"
    },
    "updated": {
      "data": "2024-08-04T06:46:30.898000+00:00",
      "provider": "mitre"
    },
    "vendors": {
      "data": [
        "redhat",
        "redhat$PRODUCT$ansible_engine",
        "redhat$PRODUCT$ansible_tower"
      ],
      "providers": [
        "nvd",
        "redhat"
      ]
    },
    "weaknesses": {
      "data": [
        "CWE-22"
      ],
      "providers": [
        "mitre",
        "nvd",
        "redhat"
      ]
    }
  },
  "redhat": {
    "cpes": [
      "cpe:/a:redhat:ansible_engine:2.7::el7",
      "cpe:/a:redhat:ansible_engine:2.8::el7",
      "cpe:/a:redhat:ansible_engine:2.8::el8",
      "cpe:/a:redhat:ansible_engine:2.9::el7",
      "cpe:/a:redhat:ansible_engine:2.9::el8",
      "cpe:/a:redhat:ansible_engine:2::el7",
      "cpe:/a:redhat:ansible_engine:2::el8",
      "cpe:/a:redhat:ansible_tower:3.4::el7",
      "cpe:/a:redhat:ansible_tower:3.5::el7"
    ],
    "created": "2020-02-18T00:00:00+00:00",
    "description": "A flaw was found in Ansible 2.7.17 and prior, 2.8.9 and prior, and 2.9.6 and prior when using the Extract-Zip function from the win_unzip module as the extracted file(s) are not checked if they belong to the destination folder. An attacker could take advantage of this flaw by crafting an archive anywhere in the file system, using a path traversal. This issue is fixed in 2.10.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {},
      "cvssV3_1": {
        "score": 7.5,
        "vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:C/C:H/I:H/A:H"
      },
      "threat_severity": "Moderate"
    },
    "redhat_repo_path": "2020/CVE-2020-1737.json",
    "references": [
      "https://nvd.nist.gov/vuln/detail/CVE-2020-1737",
      "https://www.cve.org/CVERecord?id=CVE-2020-1737"
    ],
    "title": "ansible: Extract-Zip function in win_unzip module does not check extracted path",
    "updated": "2020-02-18T00:00:00+00:00",
    "vendors": [
      "redhat",
      "redhat$PRODUCT$ansible_engine",
      "redhat$PRODUCT$ansible_tower"
    ],
    "weaknesses": [
      "CWE-22"
    ]
  }
}